src/Symfony/Bundle/FrameworkBundle/Resources/config/web.php

```diff
@@ -63,14 +63,24 @@
             ])
             ->tag('controller.argument_value_resolver', ['priority' => 100, 'name' => DateTimeValueResolver::class])
 
-        ->set('argument_resolver.request_payload', RequestPayloadValueResolver::class)
+        ->set('request_payload_value_resolver', RequestPayloadValueResolver::class)
             ->args([
                 service('serializer'),
                 service('validator')->nullOnInvalid(),
                 service('translator')->nullOnInvalid(),
             ])
+            ->lazy()
```

Comment:
I made this lazy to prevent instantiating the serialize+validator+translator tuple on controllers that don't use `MapRequest*`.

Reply:
> we need this workaround with current

I had to do this because I needed to register the service as both a listener and a resolver, which is something that was not allowed by the way traceable resolvers were wired. I fixed so I could remove the special definitions.